### PR TITLE
Feat: Function/upload data file

### PR DESCRIPTION
**Pull request by AdarPascal**

### CODE REVIEW

File: UploadDataFile/function.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
+  "bindings": [
+    {
+      "authLevel": "anonymous",
+      "type": "httpTrigger",
+      "direction": "in",
+      "name": "req",
+      "methods": ["post"],
+      "route": "dataupload/"
```
Comment (carmeltsadok) [inline UploadDataFile/function.json:9]:
```suggestion
      "route": "data-upload/"
```

File: UploadDataFile/index.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        const files = multipart.Parse(Buffer.from(req.body), multipart.getBoundary(req.headers['content-type']));
+        if (files.length > 1) throw new ValidationFunctionError(functionName, 'Multiple files upload is not supported');
+        const file = files[0];
+
+        if (!config.functions.uploadDataFiles.supportedFileTypes.includes(file.type))
+            throw new ValidationFunctionError(functionName, `[${file.type}] file type is not supported`);
+
+        if (file.type === 'application/xml') {
+            const fileAsString: string = bufferToString(file.data);
+            if (validate(fileAsString) !== true) throw new ValidationFunctionError(functionName, 'XML structure incorrect');
```
Comment (carmeltsadok) [inline UploadDataFile/index.ts:27]:
```suggestion
            if (!validate(fileAsString)) throw new ValidationFunctionError(functionName, 'XML structure incorrect');
```

File: UploadDataFile/index.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            });
+            if (filePrefixMatch.length)
+                return {
+                    status: 201,
+                    body: 'uploaded successfully',
+                };
+            throw new ValidationFunctionError(functionName, 'Unknown XML structure');
+        }
+
+        if (file.type === 'application/json') {
```
Comment (carmeltsadok) [inline UploadDataFile/index.ts:43]: change to `else if`

File: UploadDataFile/index.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        if (!req.body) throw new ValidationFunctionError(functionName, 'Missing request body');
+
+        const files = multipart.Parse(Buffer.from(req.body), multipart.getBoundary(req.headers['content-type']));
+        if (files.length > 1) throw new ValidationFunctionError(functionName, 'Multiple files upload is not supported');
+        const file = files[0];
+
+        if (!config.functions.uploadDataFiles.supportedFileTypes.includes(file.type))
+            throw new ValidationFunctionError(functionName, `[${file.type}] file type is not supported`);
+
+        if (file.type === 'application/xml') {
```
Comment (carmeltsadok) [inline UploadDataFile/index.ts:25]: maybe change to switch case instead

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

File: UploadDataFile/index.ts (hunk 2 above)
Comment (AdarPascal) [reply UploadDataFile/index.ts:27]: Lol.. idk why I wrote it like that.. embarrassing X (

File: UploadDataFile/index.ts (hunk 3 above)
Comment (AdarPascal) [reply UploadDataFile/index.ts:43]: Don't need to.
The first `if` is ending the function lifecycle if true and the second `if` won't run.

File: UploadDataFile/index.ts (hunk 4 above)
Comment (AdarPascal) [reply UploadDataFile/index.ts:25]: Why? This is the best performance and simplified way of execution.

**Review by carmeltsadok — APPROVED** [review]

**Review by yubbi99 — APPROVED** [review]

**Pull request merged**